File: src/windows/magma/app/compile/type/ArrayType.c
#include "./ArrayType.h"
export class ArrayType implements Type {
	Type child;
	constructor (Type child) {
		this.child = child;
	}
}

&[I8] generate() {
	return this/*auto*/.child.generate(/*auto*/) + "[]";
}
Bool isFunctional() {
	return false/*auto*/;
}
Bool isVar() {
	return false/*auto*/;
}
&[I8] generateBeforeName() {
	return "";
}

File: src/node/magma/app/ArrayType.ts
import { Type } from "../../magma/api/Type";
class ArrayType implements Type {
	child: Type;
	constructor (child: Type) {
		this.child = child;
	}
	generate(): string {
		return this.child.generate() + "[]";
	}
	isFunctional(): boolean {
		return false;
	}
	isVar(): boolean {
		return false;
	}
	generateBeforeName(): string {
		return "";
	}
}
